backend/src/test/java/com/github/astridottmann/backend/controllers/CompareRoutesIntegrationTest.java

```diff
@@ -0,0 +1,199 @@
+                                false,
+                                new PublicTransport("publicTransport", 46.0, "longDistance", "train"),
+                                45.26)),
+                new ComparisonResults(268.93, 45.26, 223.67)
+        );
+        testCompareRoutesJson = objectMapper.writeValueAsString(testCompareRoutes);
+
+        testCompareRoutesWithoutId = new CompareRoutes(
+                null,
+                List.of(new Route("123",
+                                "Hamburg",
+                                "Frankfurt",
+                                492,
+                                1,
+                                false,
+                                new Car("car", 253.3, "petrol", "large"),
+                                268.93),
+                        new Route("456",
+                                "Hamburg",
+                                "Frankfurt",
+                                492,
+                                1,
+                                false,
+                                new PublicTransport("publicTransport", 46.0, "longDistance", "train"),
+                                45.26)),
+                new ComparisonResults(268.93, 45.26, 223.67));
+
+        //   testCompareRoutesWithoutIdJson = objectMapper.writeValueAsString(testCompareRoutesWithoutId);
+
+        testComparedJson = """
+                  [
+                            {
+                                "id": "123",
+                                "start": "Hamburg",
+                                "destination": "Frankfurt",
+                                "distance": 492,
+                                "numberOfPersons": 1,
+                                "oneWay": false,
+                                "vehicle": {
+                                    "type": "car",
+                                    "co2Emission": 253.3,
+                                    "fuel": "petrol",
+                                    "carSize": "large"
+                                },
+                                "co2EmissionRoute": 268.93
+                            },
+                            {
+                                "id": "456",
+                                "start": "Hamburg",
+                                "destination": "Frankfurt",
+                                "distance": 492,
+                                "numberOfPersons": 1,
+                                "oneWay": false,
+                                "vehicle": {
+                                    "type": "publicTransport",
+                                    "co2Emission": 46.0,
+                                    "distanceLevel": "longDistance",
+                                    "meansOfTransport": "train"
+                                },
+                                "co2EmissionRoute": 45.26
+                            }
+                        ]
+                """;
```

Comment:
instead of write plane text as json you could use the object mapper that you already have

Reply:
I had an "id-problem", when creating the expected-object. That's the reason I wrote it completely in json